File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        //insert and check map
        var map = new Map();
        for (var i = 0; i < nums.length; i++){
            var val = nums[i];
            var complement = target - val;
            if(map.has(complement)){
                var ind = map.get(complement);
                return [ind, i];
            } else {
                map.set(val, i);
            }
        }
    }
}

File: Data Structures & Algorithms/missing-number/submission-2.java
class Solution {
    public int missingNumber(int[] nums) {
        int sum = 0;
        for (int num : nums) {
            sum ^= num;
        }
        // System.out.println(sum);
        for (int i = 0; i <= nums.length; i++){
            sum ^= i;
        }
        return sum;
    }
}
